scipy/spatial/transform/rotation.py

```diff
@@ -0,0 +1,874 @@
+        quat[:, 3] = np.cos(norms / 2)
+
+        if is_single:
+            return cls(quat[0], normalized=True, copy=False)
+        else:
+            return cls(quat, normalized=True, copy=False)
+
+    @classmethod
+    def from_euler(cls, seq, angles, degrees=False):
+        """Initialize rotation from Euler angles.
+
+        Rotations in 3 dimensions can be represented by a sequece of 3
+        rotations around a sequence of axes. In theory, any three axes spanning
+        the 3D Euclidean space are enough. In practice the axes of rotation are
+        chosen to be the basis vectors.
+
+        The three rotations can either be in a global frame of reference
+        (extrinsic) or in a body centred frame of refernce (intrinsic), which
+        is attached to, and moves with, the object under rotation [1]_.
+
+        Parameters
+        ----------
+        seq : string
+            Specifies sequence of axes for rotations. Up to 3 characters
+            belonging to the set {'X', 'Y', 'Z'} for intrinsic rotations, or
+            {'x', 'y', 'z'} for extrinsic rotations. Extrinsic and intrinsic
+            rotations cannot be mixed in one function call.
+        angles : float or array_like, shape (N,) or (N, [1 or 2 or 3])
+            Euler angles specified in radians (`degrees` is False) or degrees
+            (`degrees` is True).
+            For a single character `seq`, `angles` can be:
+
+                - a single value
+                - array_like with shape (N,), where each `angle[i]`
+                  corresponds to a single rotation
+                - array_like with shape (N, 1), where each `angle[i, 0]`
+                  corresponds to a single rotation
+
+            For 2- and 3-character wide `seq`, `angles` can be:
+
+                - array_like with shape (W,) where `W` is the width of
+                  `seq`, which corresponds to a single rotation with `W` axes
+                - array_like with shape (N, W) where each `angle[i]`
+                  corresponds to a sequence of Euler angles describing a single
+                  rotation
+
+        degrees : boolean, optional
+            If True, then the given angles are assumed to be in degrees.
+            Default is False.
+
+        Returns
+        -------
+        rotation : `Rotation` instance
+            Object containing the rotation represented by the sequence of
+            rotations around given axes with given angles.
+
+        References
+        ----------
+        .. [1] `Euler angle definitions
+                <https://en.wikipedia.org/wiki/Euler_angles#Definition_by_intrinsic_rotations>`_
+        """
+        num_axes = len(seq)
+        if num_axes < 1 or num_axes > 3:
+            raise ValueError("Expected axis specification to be a non-empty "
+                             "string of upto 3 characters, got {}".format(seq))
+
+        intrinsic = (re.compile('^[XYZ]{1,3}$').match(seq) is not None)
```

Comment:
I remember Eric commented that you can do it simpler (without compile). Can you check that?